### DISCUSSION TITLE
Manipolare stringhe nei template

### DISCUSSION THREAD

**Discussion by cobradodo** [indomus/forum#10] [HUB: Home Assistant]
Ciao, ho un sensore che genero tramite una curl usando ha-dockermon.

il sensore è:

```
  - platform: command_line
    command: "curl --request POST --url http://192.168.20.1:8126/container/caddy/exec --header 'content-type: application/octet-stream' --data '{\"command\": \"caddy version\"}'"
    name: caddy_running_version

```

il risultato è il seguente: {"status":true,"result":"v2.3.0 h1:fnrqJLa3G5vfxcxmOH/+kJOcunPLhSBnjgIvjXV/QTA=\n"}

io volevo estrarre 2.3.0 ovvero la "versione" ieri cercavo di farlo con le espressioni regolari ma non mi è riuscito (speravo di farlo tutto nel command). Secondo voi ci sono metodi ad esempio manipolando la stringa nel "value_tempalte" ?

Grazie

**Comment by Luxor73** [comment]
`value_template: "{{ value_json.result.split(' ')[0] }}"`

**Reply by cobradodo** [reply]
Perfetto...Grazie!!!
Per come la volevo ho dovuto aggiungere
```
value_template: "{{ value_json.result.split(' ')[0] | replace('v','')}}"
```

devo vedere in documentazione cosa fa esattamente quello che hai scritto perchè ho banalmente copiato ma non capito

**Reply by cobradodo** [reply]
Ora l'ho capito..
Con .split hai diviso la stringa secondo un parametro " " (lo spazio) e con lo 0 hai detto di prendere il primo valore...come se l'avessi divisa in un array fatto:

result[0]=v2.3.0
result[1]=h1:fnrqJLa3G5vfxcxmOH/+kJOcunPLhSBnjgIvjXV/QTA=\n

Grazie :-)

**Reply by Luxor73** [reply]
Esatto, analisi corretta :-)
